web/themes/new_weather_theme/assets/js/components/HourlyTable.js

```diff
@@ -42,32 +42,52 @@ class HourlyTable extends HTMLElement {
   }
 
   clickScrollRight(wrapper) {
-    const cellsToRight = Array.from(
-      this.querySelectorAll("table tr:last-child td"),
-    ).filter((cell) => {
-      const diff = cell.offsetLeft - wrapper.scrollLeft;
-      return diff > 16;
-    });
+    const containerWidth = wrapper.getBoundingClientRect().width;
+    const firstColumnWidth = this.querySelector("table tr:last-child th").getBoundingClientRect().width;
+    const visibleWidth = containerWidth;
+    const rightSide = wrapper.scrollLeft + visibleWidth;
 
-    if (cellsToRight.length) {
+    const nextCol = Array.from(this.querySelectorAll("table tr:last-child td"))
+          .find(el => {
+            const left = el.offsetLeft;
+            const right = el.offsetLeft + el.offsetWidth;
+            return left <= rightSide && right >= rightSide;
+          });   
```

Comment:
Do I understand this correctly? The "next column" is the one whose left side is inside (or on the border of) the current viewable area and whose right side is outside (or on the border of) the current viewable area?

Reply:
Yes, that's correct